compiler-core/src/type_/environment.rs

```diff
@@ -148,6 +153,7 @@ impl<'a> Environment<'a> {
 #[derive(Debug)]
 pub struct ScopeResetData {
     local_values: im::HashMap<EcoString, ValueConstructor>,
+    ignored_names: HashMap<EcoString, SrcSpan>,
```

Comment:
Prefer immutable data structure when lots of copies have to be made
```suggestion
    ignored_names: im::HashMap<EcoString, SrcSpan>,
```